sqlmesh/core/context_diff.py

```diff
@@ -41,9 +41,9 @@ class ContextDiff(PydanticModel):
     """Whether the currently stored environment record is in unfinalized state."""
     create_from: str
     """The name of the environment the target environment will be created from if new."""
-    added: t.Set[str]
```

Comment:
does any of this have to change?

Reply:
Added the snapshots to the methods, so that consumers can easily interrogate information about the snapshot. In my case, the type of node.

I originally added an added, removed, modified field for each node type, but that wouldn't scale well if more types were added.